README.md

```diff
@@ -3,8 +3,132 @@ A language server for Hoon.
+
+The default options are: `-p 80 -d 0 -u http://localhost -s zod -c lidlut-tabwed-pillex-ridrup`
+
+- `-p` is the port your ship is accessible on over HTTP
+- `-d` is the delay for running didSave events
+- `-u` is the URL of your ship
+- `-s` is the `@p` of your ship (without a sig)
+- `-c` is the `+code` of the running urbit (also without a sig)
+
+## Urbit Setup
+
+You must have a fake ship running and it must be running the hoon language server.  To create and start a fake `~zod`:
+
+```
+urbit -F zod -c zod
+```
+
+In the urbit dojo, start the language server:
+
+```
+|start %language-server
+```
+
+If you need the passcode for your ship, enter this in dojo:
+
+```
++code
+```
+
+To start the same ship again in the future just run:
+
+```
+urbit zod
+```
+
+in the same directory you created it in.
+
+## Editor Setup
+
+Your code editor now needs to use `hoon-language-server` as an LSP provider.  There are plugins for common editors that add syntax highlighting, etc. and help connect to the language server.
+
+### VSCode
+
+ * [hoon-vscode](https://github.com/famousj/hoon-vscode)
+ * [hoon-assist-vscode](https://github.com/urbit/hoon-assist-vscode)
+
+### Emacs
+
+ * [hoon-mode.el](https://github.com/urbit/hoon-mode.el)
+
+### Vim
+
+ * [hoon.vim](https://github.com/urbit/hoon.vim)
+
+`hoon.vim` does not use the language server itself, but the github page describes a setup using `vim-lsp`.
+
+### Neovim
+
+Neovim users should use [hoon.vim](https://github.com/urbit/hoon.vim) with one of the following LSP setups:
+
+#### Native LSP
+
+Install [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) to simplify your local configuration.  To use the [default configuration](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.txt#hoon_ls), add the following to `init.lua`:
+
+```
+require'lspconfig'.hoon_ls.setup{}
+```
+
+To modify the default options use:
+
+```
+require'lspconfig'.hoon_ls.setup{
+  cmd = {"hoon-language-server" "-p", "80", "-d", "500", -s "zod", "-u", "http://localhost", "-c" "lidlut-tabwed-pillex-ridrup"},
+  filetypes = {"hoon"},
+  single_file_support = true
+}
+```
+
+You can include lua snippets in your `init.vim` like so:
+
+```
+lua << EOF
+require'lspconfig'.hoon_ls.setup{}
+EOF
+```
+
+Alternatively, configure native LSP manually without additional plugins:
+
+```
+-- Only `configs` must be required, util is optional if you are using the root resolver functions, which is usually the case.
+local configs = require ('lspconfig.configs')
+local util = require ('lspconfig.util')
+
+configs['hoon_ls'] = {
+  default_config = {
+    cmd = { 'hoon-language-server', "-p", "8080" },
+    filetypes = { 'hoon' },
+    single_file_support = true,
+    root_dir = function(fname)
+      return util.find_git_ancestor() or vim.loop.os_homedir()
+    end,
+  },
+  docs = {
+    description = [[
+      https://github.com/urbit/hoon-language-server
+    ]],
+  },
+}
+```
+
```

Comment:
Thanks for updating the README with all those information. I would have helped me a lot just 2 weeks ago.

*"Alternatively, configure native LSP manually without additional plugins:"* This is not quite true. The code scnipped still depeds on the plugin `lspconfig`. This is the code which was needed back wehn `lspconfig` did not have default configuration for hoon_ls. Since last week this is obsolete.

```suggestion
```

Reply:
That is basically the default configuration for hoon_ls which is now part of `lspconfig`.